### PR TITLE
make pylibtiff compatible with libtiff 4.2 and 4.3

### PR DESCRIPTION
**Pull request by lldelisle**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

I don't know if I need to rerender...

### CODE REVIEW

**Comment by lldelisle** [comment]
I know it is generally bad to restrict versions of dependencies but here because we need to manually add the header. I think it is fine... Still I don't know if it is correct to restrict dependencies for both host and run but I think yes.

**Comment by lldelisle** [comment]
@conda-forge-admin please rerender

**Comment by lldelisle** [comment]
@conda-forge-admin, please rerender

**Review by bgruening — APPROVED** [review]

File: recipe/meta.yaml
Comment (djhoese) [inline recipe/meta.yaml:30]: Shouldn't these be `<4.4` so that all versions of 4.3.x are included?

**Review by djhoese — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml
Comment (lldelisle) [reply recipe/meta.yaml:30]: For me this recipe only support 4.3.0 4.2.0 4.1.0 and below.
I checked and there is no 4.2.x nor 4.1.x but if there is a 4.3.1 it will not work.

Comment (lldelisle) [reply recipe/meta.yaml:30]: (The file `libtiff/tiff_h_4_3_0.py` must match the exact version)

Comment (djhoese) [reply recipe/meta.yaml:30]: Oh really? For some reason I thought it was more flexible than that. Sounds good then.

**Review by djhoese — APPROVED** [review]

**Comment by lldelisle** [comment]
Unfortunately it is not flexible:
https://github.com/pearu/pylibtiff/blob/a9769ec5d9533b14915d65b3e776dd95564b8d11/libtiff/libtiff_ctypes.py#L71-L73

**Comment by lldelisle** [comment]
Are we waiting for another approval or can this PR be merged?
Thanks

**Comment by lldelisle** [comment]
Thanks

**Pull request merged**
